include/kernels/madwf_tensor.cuh

```diff
@@ -0,0 +1,205 @@
+#pragma once
+
+#include <color_spinor_field.h>
+#include <color_spinor_field_order.h>
+#include <shared_memory_cache_helper.cuh>
+#include <kernels/madwf_transfer.cuh>
+#include <block_reduce_helper.h>
+
+namespace quda
+{
```

Comment:
Can you add some documentation to this file?  A general description of what the kernel does here would be welcome.

Reply:
https://github.com/lattice/quda/pull/1061/commits/cefb2a527fa7cb1303a9e7487b9fea4f39d1bf8d